refinery/tool_manager/models.py

```diff
@@ -122,15 +125,12 @@ class ToolDefinition(models.Model):
     )
 
     uuid = UUIDField(unique=True, auto=True)
-    name = models.TextField(unique=True, max_length=100, blank=False,
-                            null=False)
-    description = models.TextField(unique=True, max_length=500, blank=False,
-                                   null=False)
-    tool_type = models.CharField(max_length=100, choices=TOOL_TYPES,
-                                 blank=False, null=False)
-    file_relationship = models.ForeignKey("FileRelationship")
-    output_files = models.ManyToManyField("OutputFile")
-    parameters = models.ManyToManyField("Parameter")
+    name = models.TextField(unique=True, max_length=100)
+    description = models.TextField(unique=True, max_length=500)
```

Comment:
@hackdna sure, I'm "failing fast" now and not commiting any data upon the running of `generate_tool_definitions` command unless things look perfect. I'd like validation of workflow annotations to be polished well since I feel that with this flexibility we are giving users, we are also giving them more ways to potentially make errors.

Looking into workflow annotation validation using JSON Schema since we already have schemas defined in swaggerhub.

@jkmarx Also I just checked Galaxy and there is no enforcement of unique Workflow naming there.